torch/ao/quantization/pt2e/representation/rewrite.py

```diff
@@ -131,6 +131,33 @@ def _reference_quantized_max_pool2d(x_i8, x_scale, x_zero_point, out_scale, out_
     output_i8 = output_i8.to(torch.int8)
     return output_i8
 
+_QUANTIZED_ADAPTIVE_AVG_POOL2D_EXAMPLE_INPUTS = (
+    torch.randint(-128, 127, (1, 3, 3, 3), dtype=torch.int8),
+    torch.randn(1, dtype=torch.float),
+    torch.zeros(1, dtype=torch.int),
+    torch.randn(1, dtype=torch.float),
+    torch.zeros(1, dtype=torch.int),
+    torch.tensor([-128], dtype=torch.int),
+    torch.tensor([127], dtype=torch.int),
+)
+
+def _qdq_quantized_adaptive_avg_pool2d(x_i8, x_scale, x_zero_point, out_scale, out_zero_point, quant_min, quant_max):
+    output_size = (1, 1)
+    x_fp32 = torch.ops.quantized_decomposed.dequantize_per_tensor(x_i8, x_scale, x_zero_point, quant_min, quant_max, torch.int8)
+    out_fp32 = torch.ops.aten.adaptive_avg_pool2d(x_fp32, output_size)
+    out_i8 = torch.ops.quantized_decomposed.quantize_per_tensor(out_fp32, out_scale, out_zero_point, quant_min, quant_max, torch.int8)
+    return out_i8
+
+def _reference_quantized_adaptive_avg_pool2d(x_i8, x_scale, x_zero_point, out_scale, out_zero_point, quant_min, quant_max):
+    output_size = (1, 1)
+    x_i32 = x_i8.to(torch.int32)
+    # TODO: use mul.Scalar, need to change how literal args are handled
+    x_i32 = out_dtype(torch.ops.aten.mul.Tensor, torch.int32, x_i32, (x_scale / out_scale))
+    acc_i32 = torch.ops.aten._adaptive_avg_pool2d(x_i32, output_size)
+    acc_i32 = acc_i32 - x_zero_point * x_scale / out_scale + out_zero_point # int32 constant
```

Comment:
As my comment on maxpool2d diff, this should be rewritten such that
1. zero point subtract
2. adaptive_avg_pool
3. rescale with x_cale/out_scale
4. add zero point.